src/pg4n/psqlwrapper.py

```diff
@@ -1,127 +1,163 @@
-"""Interface with psql and capture all input and output."""
+"""Interface with psql and capture all input and output.
 
-from typing import Callable, List, TextIO
-from functools import reduce
+Uses pexpect in combination with pyte for interfacing and screen-scraping
+"""
+
+from typing import Callable, List
 import pexpect
+from pyte import Stream, Screen
+from shutil import get_terminal_size
 
 from .psqlparser import PsqlParser
 
-# ANSI escape codes (\x1b[) used by psql:
-# K clear part of the line
-# m SGR (set graphics mode)
-# ?1049 these
-# ?2004 are
-# ?1h   cursor?
-# ?1l   escapes (h = high, l = low)
-# >
-# =
-
 
 class PsqlWrapper:
     """Handles terminal interfacing with psql, using the parameter parser \
-    to pick up relevant SQL statements for the hook function."""
+    to pick up relevant SQL statements for the hook function.
+
+    Only one (static) class instance is intended for use in program.
+    """
 
     debug: bool = False
-    fout: TextIO
 
     input_log: bytes = b''
     output_log: bytes = b''
     db_name: bytes = b''
-    analyze: Callable[str, str]
+    analyze: Callable[[str], str]
```

Comment:
Tuolla rakentajassa (`__init__`) määritellään jäsenmuuttujia (`self.varname=...`), mikä ei näihin staattisiin muutujiin koske:
```python
>>> class C:  
...  x=0
...  def __init__(self):
...   self.x=42
... 
>>> c=C()
>>> c.x
42
>>> C.x
0
```
Pitää viitata luokkaan (rakentajassa `class.varname=...`:
```python
>>> class C:
...  x=0
...  def __init__(self):
...   C.x=42
... 
>>> c=C()
>>> c.x
42
>>> C.x
42
```
Jos halutaan manipuloida staattisia muuttujia, niin silloin on ehkä mielekästä evätä moni-ilmentymäisyys, esim. käyttäen singleton mallia.
https://stackoverflow.com/questions/6760685/creating-a-singleton-in-python

Reply:
Joo, siihen onkin unohtunut aika sekavasti nuo. Voisin poistaa nuo staattiset muuttujat ja lisätä `__init__`iin type hintit.